### DISCUSSION TITLE
RAG : display document source in the citation (feature)

### DISCUSSION THREAD

**Discussion by mlier** [open-webui/open-webui#3326] [Ideas]
Hello,
In the "documents" menu, I have added some documents in a category which name is, for instance, "myCat".
I have created a chatbot (model) where I added documents using my category "myCat".
When I do a request the chatbot give the answer and just after the citations like : 1 myCat   2 myCat   3 myCat...

Is it possible to know the name of the exact document for the citation and not only the name of the category ?

**Comment by justinh-rahb** [comment]
Add the individual documents instead of the category to the model documents configuration.

**Reply by mlier** [reply]
It is exactly what I have just done. It works. Thanks.

My second problem is that I have nearly 100 documents to add that is long to do and all the documents are displayed on one line on the interface.
It is then very difficult to check that I have added all the documents.

**Reply by justinh-rahb** [reply]
Possible UX enhancement we could make here would be when a category tag is selected here we have it automatically add the underlying individual documents, and display them in multiple rows if overflowing?

**Reply by mlier** [reply]
Yes it would be very nice 👍
